Point email footer links to correct sites

monfresh opened the pull request:
**Why**: Up until now, we had placeholder links in the email footer
because we didn't have sites to point them to. Now that we do, we need
to update the links.

Note that the https://login.gov/policy link doesn't seem to be live yet.
I'm not sure if that's the wrong URL or if the page hasn't been
published yet.

zachmargolis commented on app/views/layouts/user_mailer.html.slim:
```diff
@@ -94,9 +94,11 @@ html xmlns="http://www.w3.org/1999/xhtml"
                                           p.text-center
-                                            a href="#" = t('mailer.about', app: APP_NAME)
+                                            a href="#{MarketingSite.base_url}"
+                                              = t('mailer.about', app: APP_NAME)
```
No need to quote this IIRC:

```
a href=MarketingSite.base_url
```

zachmargolis commented on app/services/marketing_site.rb:
```diff
@@ -1,6 +1,10 @@
   BASE_URL = URI('https://login.gov').freeze
 
+  def self.base_url
+    BASE_URL
```
Let's `.to_s` this for consistency?

jessieay replied:
anything in a view will have `to_s` called on it by default (HELLO RAILS ✨ ). 

I think you know this and so you are suggesting that we call `to_s` here so we  know in our tests we are testing the correct behavior? Want to make sure I understand

jessieay commented on app/views/layouts/user_mailer.html.slim:
```diff
@@ -94,9 +94,11 @@ html xmlns="http://www.w3.org/1999/xhtml"
                                         th
                                           p.text-center
-                                            a href="#" = t('mailer.about', app: APP_NAME)
+                                            a href="#{MarketingSite.base_url}"
```
thoughts on using a `link_to` helper instead?

monfresh replied:
That's what I went for at first but couldn't get it to work for some reason. The email wouldn't even get sent! If you know how, please let me know. I'll try it again too.

monfresh commented:
@jessieay The `link_to` is working now. Not sure what happened the first time.
@zachmargolis I removed the quotes and added `to_s` and a service spec.
PTAL.

zachmargolis commented:
@jessieay re: `to_s` I just wanted to make sure we had strings, because URIs look and act almost like strings, but not quite, and I figure strings are more common to pass around? I guess I'd be OK if we switched them all to URI objects, but at the very least I think everything coming out of `MarketingSite` should be consistent
